client/src/views/Party/PlayAgain.jsx

```diff
@@ -1,29 +1,36 @@
 import React, { useState } from "react";
 
 import api from "../../utils/api";
-import { TextInput, Button } from "../../components";
+import { TextInput, Button, Checkbox } from "../../components";
 
 function PlayGame({ party, onPlayAgainClose }) {
+  const [keepSameTeams, setKeepSameTeams] = useState(false);
   const [teamsCount, setTeamsCount] = useState(party.settings.teamsCount);
   const [rotations, setRotations] = useState(party.settings.rotations);
+  const [isSubmitting, setIsSubmitting] = useState(false);
   const [turnDurationSeconds, setTurnDurationSeconds] = useState(
     party.settings.turnDurationSeconds
   );
 
   const handlePlayAgainClick = async () => {
+    setIsSubmitting(true);
+
     try {
       await api.updateSettings({
         slug: party.slug,
         settings: {
-          teamsCount,
+          ...(!keepSameTeams && { teamsCount }),
```

Comment:
Why is this necessary?

Reply:
probably would decent idea for me to add a comment here. 

But this makes so you can only change the teamsCount (i.e. the number of teams) if keepSameTeams is not selected. If you had a different number of teams you couldn't keep the same teams. 

This is a pattern I end up using a lot. Looks weird but is a nice way of conditionally add an entry to an object. 

The API is set-up so it will only update the fields that are passed to backend.